src/modbusrtu.cpp

```diff
@@ -4,3 +4,96 @@
  *  This source code is licensed under the MIT license found in the
  *  LICENSE file in the root directory of this source tree.
  */
+
+#include "modbusrtu.h"
+
+#include <stdio.h>
+
+// Linux headers
+#include <fcntl.h>      // Contains file controls like O_RDWR
+#include <errno.h>      // Error integer and strerror() function
+#include <termios.h>    // Contains POSIX terminal control definitions
+#include <unistd.h>     // write(), read(), close()
+
+using namespace osdev::components::modbus;
+
+ModbusRtu::ModbusRtu( const ConnectionConfig &conf )
+    : m_conConfig( conf )
+    , m_socket( -1 )
+{
+}
+
+bool ModbusRtu::Connect()
+{
+    m_socket = open( m_conConfig.getPortName().c_str(), O_RDWR);
```

Comment:
Shall I nitpick about the wide variance in whitespace usage, or shall I just ignore that? :P

(space here between `(` and `m_conConfig`, line 29 has no spaces around `()`, line 37 has, etc etc :P

Reply:
I see what you mean. Force of habit..